### PR TITLE
#86 Add test with 0 element but not empty

### LINKED ISSUE #86: [Proposal]Add test with 0 element but not empty
### Proposal.

Add common test case for empty element but no emptyIterable.  
The test checks for no panic.
Use test source like below.

```golang
itb := gcf.FromSlice([]int{1})
itb = gcf.Filter(itb, func(v int) bool { return false })
```

### What benefit with the proposal?

Since each Iterator does not expect to pass 0 Iterators, it could panic trying to process 0 Iterators.  
By passing this test prevent that panic.

### Others

> emptyじゃない時にパニック起こさない汎用のテストケース作ってソース以外の関数に適用する

_Originally posted by @meian in https://github.com/meian/gcf/pull/83#discussion_r799507648_

### PR DESCRIPTION
**Pull request by meian**
Closes #86

### What changed

- Add empty but not emptyIterable test case. 
- Rename testEmptyChain to testEmpties
    - The timing of running the new test case is the same as the existing, so rename to be matched to both case.  

### Others

----

### Checklist

- [x] **Issue is linked.**
    - It is not needed if simple edit such as corrections of words, etc.
- [x] **Test is added.**
    - If fixed bug, add case on existing test or add new test.
    - If add new feature, add new test.
    - Add benchmark test, as needed.
- [x] **Test coverage rate has no decreased.**
    - If rate is decreased, explain why in What changed.

### CODE REVIEW

File: common_test.go [hunk 1]
```diff
@@ -25,13 +25,29 @@ func testBeforeAndAfter[T any](t *testing.T, itb gcf.Iterable[T]) {
...
+func testEmpties(t *testing.T, f func(itb gcf.Iterable[int]) gcf.Iterable[int]) {
 	t.Helper()
-	itbe := gcf.FromSlice([]int{})
-	t.Run("is empty Iterable", func(t *testing.T) {
-		itb := f(itbe)
+	t.Run("emptyIterable chaining", func(t *testing.T) {
+		itb := gcf.FromSlice([]int{})
+		itb = f(itb)
 		assert.True(t, gcf.IsEmptyIterable(itb), "%v", gcf.ToSlice(itb))
 	})
```
Comment (meian) [inline common_test.go:40]: 既存の emptyIterable からチェーンすると emptyIterable になるテスト

File: common_test.go [hunk 2]
```diff
@@ -25,13 +25,29 @@ func testBeforeAndAfter[T any](t *testing.T, itb gcf.Iterable[T]) {
...
+			err := recover()
+			if err != nil {
+				t.Errorf("%v", err)
+			}
+		}()
+		itb := gcf.FromSlice([]int{1})
+		itb = gcf.Filter(itb, func(v int) bool { return false })
+		itb = f(itb)
+		_ = gcf.ToSlice(itb)
+	})
```
Comment (meian) [inline common_test.go:52]: IteratorがemptyIterator以外を渡されて要素数が0になるケースを想定していないことがあっったため、その際にパニックが起きないことを確認するためのテスト

File: common_test.go [hunk 3]
```diff
@@ -25,13 +25,29 @@ func testBeforeAndAfter[T any](t *testing.T, itb gcf.Iterable[T]) {
...
 	})
+	t.Run("no panic from empty", func(t *testing.T) {
+		defer func() {
+			err := recover()
+			if err != nil {
+				t.Errorf("%v", err)
+			}
+		}()
+		itb := gcf.FromSlice([]int{1})
+		itb = gcf.Filter(itb, func(v int) bool { return false })
```
Comment (meian) [inline common_test.go:49]: 空だけどemptyIterableでない旨をコメント振っておいた方がよさそう
